src/schedule.py

```diff
@@ -86,20 +88,97 @@ def cfg_to_dfg(cfg, graphs, latency):
 
     return cfg_node_to_dfg_map
 
+def assign_time_of_execution(graph):
+    """
+    Calculates when each operation will take place,
+    overwriting the "dist" attribute from before.
+    """
+    for node in graph.nodes:
+        graph.nodes[node]["dist"] =  0
+    graph = nx.reverse(graph)
+    for gen in list(nx.topological_generations(graph))[1:]:
+        max_weight = 0
+        for node in gen:
+            in_edge = list(graph.in_edges(node))[0]
+            max_weight = max(max_weight, graph.edges[in_edge]["weight"])
+        for node in gen:
+            in_edge = list(graph.in_edges(node))[0]
```

Comment:
Shouldn't these be out edges since you're operating on the reversed graph

Reply:
my understanding was that out edges correspond to latency of nodes in the forward graph, so in the reverse graph it will be flipped. is that not right?